src/audio-manager.ts

```diff
@@ -0,0 +1,279 @@
+import {_audioContext} from './audio-listener.js';
+
+/**
+ * AudioManager loads and manages audio files from which PlayableNodes are created.
+ *
+ * @example
+ * ```js
+ * async start() {
+ *      this.audio = await audioManager.load('click.wav');
+ * }
+ * onPress() {
+ *      this.audio.play();
+ * }
+ * // if not needed any longer
+ * this.audio.destroy();
+ * ```
+ */
+export class AudioManager {
+    private _bufferCache: Map<string, [AudioBuffer, number]> = new Map();
+
+    /**
+     * Creates a PlayableNode from provided audio file.
+     *
+     * @param source Path to the file from which to create a PlayableNode.
+     */
+    async load(source: string): Promise<PlayableNode> {
+        return new PlayableNode(source, await this._add(source), this);
+    }
+
+    /**
+     * Adds the specified file to cache.
+     * @param source Path to the file that should be added to cache.
+     * @warning This is for internal use only, use at own risk!
+     */
+    async _add(source: string): Promise<AudioBuffer> {
```

Comment:
If you make it private, it's not usable from the outside easily in TypeScript, at least the compiler complains.